tests/containers/host_configuration.pm

```diff
@@ -12,26 +12,53 @@
 use Mojo::Base qw(consoletest);
 use testapi;
 use utils;
-use Utils::Systemd qw(disable_and_stop_service);
-use version_utils qw(check_os_release is_sle);
+use version_utils qw(check_os_release get_os_release is_sle);
+use containers::common;
 
 sub run {
     my ($self) = @_;
     $self->select_serial_terminal;
     my $interface;
-    if (check_os_release('suse', 'PRETTY_NAME')) {
-        $interface = script_output q@ip r s default 0.0.0.0/0 | awk '{printf $5}'@;
-        validate_script_output "ip a s '$interface'", sub { m/((\d{1,3}\.){3}\d{1,3}\/\d{1,2})/ };
+    my $update_timeout = 600;
+    my ($version, $sp, $host_distri) = get_os_release;
+    my $engine = get_required_var('CONTAINER_RUNTIME');
+
+    # Update the system to get the latest released state of the hosts.
+    # Check routing table is well configured
+    if ($host_distri eq 'sles') {
+        zypper_call("--quiet up", timeout => $update_timeout);
         ensure_ca_certificates_suse_installed();
-        disable_and_stop_service(opensusebasetest::firewall, ignore_failure => 1);
     }
     else {
-        # Re-running dhclient on RHEL is confusing the routing tables
-        assert_script_run "dhclient -v" unless get_var("HDD_1") =~ /rhel/;
-        $interface = script_output q@ip r s default | head -1 | awk '{printf $5}'@;
-        validate_script_output "ip a s '$interface'", sub { m/((\d{1,3}\.){3}\d{1,3}\/\d{1,2})/ };
-        assert_script_run "curl http://ca.suse.de/certificates/ca/SUSE_Trust_Root.crt -o /etc/ssl/certs/SUSE_Trust_Root.crt" if is_sle();
+        if ($host_distri eq 'ubuntu') {
+            assert_script_run("dhclient -v");
+            assert_script_run("apt-get update -qq -y", timeout => $update_timeout);
+        } elsif ($host_distri eq 'centos') {
+            assert_script_run("dhclient -v");
+            assert_script_run("yum update -y --allowerasing", timeout => $update_timeout);
+        } elsif ($host_distri eq 'rhel') {
+            assert_script_run("yum update -y", timeout => $update_timeout);
+        }
+    }
+
+    # Make sure we can access internet and DNS works
+    script_retry('ping -c 3 www.google.com');
+
+    # Install engines in case they are not installed
+    install_docker_when_needed($host_distri) if ($engine =~ 'docker');
```

Comment:
using -eq instead of =~? only a suggestion :)

Reply:
no, because `$engine` can take as value `docker,podman`

Reply:
Oh yes! thank you